lib/server/exashareresolver.py

```diff
@@ -32,19 +32,15 @@ def supports(url):
 
 
 def resolve(url):
-    data = re.search(r'<script[^\.]+?\.setup\((.+?)\);', util.request(url), re.I | re.S)
+    realurl = re.search(r'<iframe src="([^"]+)".*', util.request(url), re.I | re.S).group(1)
+    data = re.search(r'<script[^\.]+?\.setup\((.+?)\);', util.request(realurl), re.I | re.S)
     if data:
         data = data.group(1).decode('string_escape')
-        data = re.sub(r'\w+\(([^\)]+?)\)', r'\1', data)  # Strip JS functions
+        data = re.sub(r'\w+\(([^\)]+?)\)', r'\1', data).replace('primaryCookie','"primaryCookie"')  # Strip JS functions
         data = demjson.decode(data)
-        if 'playlist' in data:
+        if 'sources' in data:
             result = []
-            for stream in data['playlist']:
-                if 'tracks' in stream:
-                    for track in stream['tracks']:
-                        result.append({'url': stream['file'], 'subs': track['file'],
-                                       'lang': ' %s subtitles' % track['label']})
```

Comment:
i haven't yet come across a stream with subtitles so i don't really have a test-case. anyway, it should be easy to bring them back once there's something to test with.